examples/api/lib/material/date_picker/show_date_picker.1.dart

```diff
@@ -0,0 +1,67 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+/// Flutter code sample for basic [showDatePicker].
+
+void main() => runApp(const DatePickerApp());
+
+class DatePickerApp extends StatelessWidget {
+  const DatePickerApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: Scaffold(
+        appBar: AppBar(title: const Text('showDatePicker Example')),
+        body: const Center(child: DatePickerExample()),
+      ),
+    );
+  }
+}
+
+class DatePickerExample extends StatefulWidget {
+  const DatePickerExample({super.key});
+
+  @override
+  State<DatePickerExample> createState() => _DatePickerExampleState();
+}
+
+class _DatePickerExampleState extends State<DatePickerExample> {
+  DateTime? selectedDate;
+
+  Future<void> _selectDate() async {
+    final DateTime? pickedDate = await showDatePicker(
+      context: context,
+      initialDate: DateTime(2021, 7, 25),
+      firstDate: DateTime(2021),
+      lastDate: DateTime(2022),
+    );
+    if (mounted && pickedDate != null && pickedDate != selectedDate) {
+      setState(() {
+        selectedDate = pickedDate;
+      });
+    }
```

Comment:
```suggestion
    setState(() {
      selectedDate = pickedDate;
    });
```

This seems way too complicated than neccesary.

Reply:
Thanks for bringing this up! I can share the if-statement's rationale:

- `mounted`: Flutter currently has a [use_build_context_synchronously](https://dart.dev/lints/use_build_context_synchronously) lint to prevent a situation where a `context` is referenced after being disposed of. There's also a [proposal](https://github.com/dart-lang/linter/issues/4892) to have the same apply to `setState` calls as well.
- `pickedDate != null`: If the user taps "OK", then the future returned by `showDatePicker` completes with the selected DateTime value; otherwise its value is `null`. Including the null check in the if-statement ensures that tapping outside the date picker doesn't remove the previous selection.
- `pickedDate != selectedDate`: If these two are equal, then rebuilding isn't necessary.

The `mounted` check won't ever be false in this example app, but I feel that our documentation should still encourage this coding practice.

Reply:
1. Regarding `mounted`, I'm not sure if it used when the example  has no issues that could benefit from it. When someone is reading this sample might get lost what it does without the level experience you've. They might think mounted is necessary.

2. `pickedDate != null`, if the user presses ok without selection, they're effectively not selecting a date so it the `selectedDate` should be reset IMO. This is an example so they can figure out the logic to not reset the date if they want. The goal of the example is to show how the widget works. Adding complex logic that might seems like it is required when using the widget. 

3. `pickedDate != selectedDate`, setState rebuilds is a common Flutter knowledge. Not sure if it is necessary in a simple widget sample.